pkg/analytics/executions.go

```diff
@@ -0,0 +1,95 @@
+package analytics
+
+import (
+	"github.com/bacalhau-project/bacalhau/pkg/models"
+)
+
+const ExecutionEventType = "bacalhau.execution_v1"
+
+type ExecutionEvent struct {
+	ExecutionID string `json:"execution_id,omitempty"`
+	JobID       string `json:"job_id,omitempty"`
+	EvalID      string `json:"eval_id,omitempty"`
+
+	// name of execution
+	Name string `json:"name,omitempty"`
+	// hash of the node name running execution
+	NodeNameHash string `json:"node_name_hash,omitempty"`
+	// hash of namespace execution is running in
+	NamespaceHash string `json:"namespace_hash,omitempty"`
+
+	// AllocatedResources
+	// NB: total execution resources for all tasks in execution
+	TotalCPUUnits    float64 `json:"total_cpu_units,omitempty"`
+	TotalMemoryBytes uint64  `json:"total_memory_bytes,omitempty"`
+	TotalDiskBytes   uint64  `json:"total_disk_bytes,omitempty"`
+	TotalGPUCount    uint64  `json:"total_gpu_count,omitempty"`
+
+	DesiredState        string `json:"desired_state,omitempty"`
+	DesiredStateMessage string `json:"desired_state_message,omitempty"`
+	ComputeState        string `json:"compute_state,omitempty"`
+	ComputeStateMessage string `json:"compute_state_message,omitempty"`
```

Comment:
as discussed yesterday, lets just include the error code with this event, and have a separate event with raw messages. The other event can just have the execution id and raw messages if you want to keep things simple. We should only publish those when the execution fails or rejected as we don't care about tracking those for successful executions

Reply:
> lets just include the error code with this event

What field is the error code? Do we mean the `ExitCode` from `RunCommandResult`?

Reply:
No its is bacerrors.ErrorCode. Unfortunately I just realized we are not adding that error code to `models.Event` and not propagating it back to the orchestrator. We can do this in v1.5.1 I guess